frontend/src/scenes/onboarding/onboardingLogic.tsx

```diff
@@ -347,6 +347,10 @@ export const onboardingLogic = kea<onboardingLogicType>([
             if (productKey !== values.productKey) {
                 actions.setProductKey(productKey)
             }
+
+            // Reset onboarding steps so they can be populated upon render in the component.
+            actions.setAllOnboardingSteps([])
```

Comment:
Are we at risk here for some race condition where hitting this page resets them, but the component tries to set them, and it might happen out of order?

Reply:
I don't think that's possible. The component sets the steps in a useEffect, which by design runs after render. These steps should definitely happen before the component updates state.